tools/ci_connector_ops/ci_connector_ops/pipelines/connectors_ci.py

```diff
@@ -0,0 +1,289 @@
+    test_context.updated_secrets_dir = None
+    return test_context
+
+
+async def teardown(test_context: ConnectorTestContext, test_report: ConnectorTestReport) -> ConnectorTestContext:
+    """Finish pipeline execution by saving updated secrets to GSM and upload test reports to S3.
+
+    Args:
+        test_context (ConnectorTestContext): The context for a connector specific test pipeline.
+        test_report (ConnectorTestReport): The test report to upload to S3.
+
+    Returns:
+        ConnectorTestContext: The connector test context.
+    """
+    teardown_pipeline = test_context.dagger_client.pipeline(f"Teardown {test_context.connector.technical_name}")
+    if test_context.should_save_updated_secrets:
+        await secrets.upload(
+            teardown_pipeline,
+            test_context.connector,
+            test_context.updated_secrets_dir,
+        )
+
+    test_context.logger.info(str(test_report))
+    local_test_reports_path_root = "tools/ci_connector_ops/test_reports/"
+
+    connector_name = test_report.connector_test_context.connector.technical_name
+    connector_version = test_report.connector_test_context.connector.version
+    git_revision = test_report.connector_test_context.git_revision
+    git_branch = test_report.connector_test_context.git_branch.replace("/", "_")
+    suffix = f"{connector_name}/{git_branch}/{connector_version}/{git_revision}.json"
+    local_report_path = Path(local_test_reports_path_root + suffix)
+    local_report_path.parents[0].mkdir(parents=True, exist_ok=True)
+    local_report_path.write_text(test_report.to_json())
+    if test_report.should_be_saved:
+        s3_reports_path_root = "python-poc/tests/history/"
+        s3_key = s3_reports_path_root + suffix
+        await remote_storage.upload_to_s3(teardown_pipeline, str(local_report_path), s3_key, os.environ["TEST_REPORTS_BUCKET_NAME"])
+    if not test_context.is_local:
+        send_commit_status_check(
+            test_context.git_revision,
+            "success" if test_report.success else "failure",
+            test_context.gha_workflow_run_url,
+            f"Finished tests for {test_context.connector.technical_name}",
+        )
+    return test_context
+
+
+async def run(test_context: ConnectorTestContext) -> ConnectorTestReport:
+    """Runs a CI pipeline for a single connector.
+    A visual DAG can be found on the README.md file of the pipelines modules.
+
+    Args:
+        test_context (ConnectorTestContext): The initialized test context.
+
+    Returns:
+        ConnectorTestReport: The test reports holding tests results.
+    """
+    test_context = await setup(test_context)
+    qa_checks_results_future = asyncio.create_task(tests.run_qa_checks(test_context))
+    connector_source_code = await environments.with_airbyte_connector(test_context, install=False)
+    connector_under_test = await environments.with_airbyte_connector(test_context)
```

Comment:
I think we can improve the readability (and maybe performance as well) on the body of this coroutine. I'll see if I can come up with something tomorrow and get back to you.